x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/registry/ModelRegistry.java

```diff
@@ -662,163 +661,261 @@ public void storeModel(Model model, ActionListener<Boolean> listener, TimeValue
     }
 
     private void storeModel(Model model, boolean updateClusterState, ActionListener<Boolean> listener, TimeValue timeout) {
-        ActionListener<BulkResponse> bulkResponseActionListener = getStoreIndexListener(model, updateClusterState, listener, timeout);
-        String inferenceEntityId = model.getConfigurations().getInferenceEntityId();
-        var configRequestBuilder = createIndexRequestBuilder(
-            inferenceEntityId,
-            InferenceIndex.INDEX_NAME,
-            model.getConfigurations(),
-            false,
-            client
-        );
-        var secretsRequestBuilder = createIndexRequestBuilder(
-            inferenceEntityId,
-            InferenceSecretsIndex.INDEX_NAME,
-            model.getSecrets(),
-            false,
-            client
-        );
+        storeModels(List.of(model), updateClusterState, listener.delegateFailureAndWrap((delegate, responses) -> {
+            var firstFailureResponse = responses.stream().filter(ModelStoreResponse::failed).findFirst();
+            if (firstFailureResponse.isPresent() == false) {
+                delegate.onResponse(Boolean.TRUE);
+                return;
+            }
+
+            var failureItem = firstFailureResponse.get();
+            if (ExceptionsHelper.unwrapCause(failureItem.failureCause()) instanceof VersionConflictEngineException) {
+                delegate.onFailure(
+                    new ElasticsearchStatusException(
+                        "Inference endpoint [{}] already exists",
+                        RestStatus.BAD_REQUEST,
+                        failureItem.failureCause(),
+                        failureItem.inferenceId
```

Comment:
Nitpick, but there's inconsistent use of getter methods vs. directly accessing the field for this object.